shared/src/main/java/uk/gov/di/audit/AuditContext.java

```diff
@@ -0,0 +1,160 @@
+package uk.gov.di.audit;
+
+import uk.gov.di.authentication.shared.services.AuditService.MetadataPair;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class AuditContext {
```

Comment:
This can be a record (in general the only things that I'd still make classes are things that encapsulate e.g. services and lambdas and dynamo db entities which aren't compatible with records yet)

Reply:
Ah sorry I see that you're updating fields on this as you go, and the syntax for doing this in records isn't particularly nice yet. For consistency in terms of refactoring this, I'd probably want to go for the same approach we've done in the main api that instead constructs an audit user since this is the thing that is eventually used in the audit service anyway - see [here](https://github.com/govuk-one-login/authentication-api/pull/4402). 

If this ends up being a bigger conversation and holds this work back too much, maybe we can just introduce the changes we need for this work in this PR, and refactor separately.